include/deal.II/matrix_free/mapping_info.h

```diff
@@ -441,6 +441,25 @@ namespace internal
        */
       SmartPointer<const Mapping<dim>> mapping;
 
+      /**
+       * Internal function to compute the geometry for the case the mapping is
+       * a MappingQ and a single quadrature formula per slot (non-hp case) is
+       * used. This method computes all data from the underlying cell
+       * quadrature points using the fast operator evaluation techniques from
+       * the matrix-free framework itself, i.e., it uses a polynomial
+       * description of the cell geometry (that is computed in a first step)
+       * and then computes all Jacobians and normal vectors based on this
+       * information. This optimized approach is much faster than going
+       * through FEValues and FEFaceValues, especially when several different
+       * quadrature formulas are involved, and consumes less memory.
+       */
+      void
+      compute_mapping_q(
+        const dealii::Triangulation<dim> &                        tria,
+        const std::vector<std::pair<unsigned int, unsigned int>> &cells,
+        const std::vector<
+          FaceToCellTopology<VectorizedArrayType::n_array_elements>> &faces);
```

Comment:
Documentation of the parameters. I am wondering what the meaning of `cells` is (interior & exterior)?

Reply:
Is there also a function for cells?

Reply:
I guess the vector is for all faces. Any chance to provide a method, which only works on a macro cells so that we could call this function to the mapping on-the-fly class.

Reply:
I see what you mean but I think this is wrong place to ask for that - this function does the whole initialization both the cells and the faces and all the other tricks we apply like looking for compressions. What we are looking for for the mapping on the fly is the things I put into the `ExtractCellHelper::compute_range_mapping_q` and `ExtractFaceHelper::compute_range_mapping_q`, or, more precisely, the body inside the loop over cell batches there. I think we should reorganize things a bit for that purpose, but I wanted to keep the generalizations (like adding the `fe_degree` and `n_q_points_1d` templates for the case we know them) out of this PR because it is big enough already. In any case, I think now we have collected the code that is needed, so we can move it around later when we use it for different purposes.

Reply:
I agree. I put the issue onto my TODO list!